### PR TITLE
feat(CI): Reduce usage of github API

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Use direct informations available in github events rather than API

### Motivation
Leverage github action avantages and reduce pressure on API rate limit

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .github/workflows/label-analysis.yml [hunk 1]
```diff
@@ -13,39 +13,45 @@ jobs:
...
-          pip install -r requirements.txt
-          pip install -r tasks/libs/requirements-github.txt
-      - name: Run PR check for team assignment label
-        env:
-          BRANCH_NAME: ${{ github.ref_name }}
-          PR_ID: ${{ github.event.pull_request.number }}
-        run: |
-          inv -e lint-teamassignment
+          labels="${{join(github.event.pull_request.labels.*.name, ' ')}}"
+          for label in $(echo "$labels" | tr ' ' '\n'); do
```
Comment (chouquette) [inline .github/workflows/label-analysis.yml:19]: nit: I don't think you need th ` | tr` part
`for label in $labels` should work out of the box

File: .github/workflows/label-analysis.yml [hunk 2]
```diff
@@ -13,39 +13,45 @@ jobs:
...
+              skip=1
+            fi
+            if [[ "$label" == "qa/done" ]]; then
+              done=1
+            fi
+            if [[ "$label" == "qa/no-code-change" ]]; then
+              noChange=1
+            fi
+          done
+          if [[ ($skip -eq 0 && $done -eq 0 && $noChange -eq 0) || ($skip -eq 1 && $((done ^ noChange)) -eq 1) ]]; then
```
Comment (chouquette) [inline .github/workflows/label-analysis.yml:52]: I think `($skip -eq 1 && $((done ^ noChange)) -eq 1)` could be written as `$skip -eq 1 && $done -neq $noChange` which is more readable IMO

File: .github/workflows/milestone-analysis.yml [hunk 3]
```diff
@@ -12,19 +12,14 @@ jobs:
...
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          pip install -r tasks/libs/requirements-github.txt
-      - name: Run PR check for milestone assignment
-        env:
-          BRANCH_NAME: ${{ github.ref_name }}
-          PR_ID: ${{ github.event.pull_request.number }}
-        run: |
-          inv -e lint-milestone
+          milestones="${{join(github.event.pull_request.milestone.*.name, ' ')}}"
```
Comment (chouquette) [inline .github/workflows/milestone-analysis.yml]: Can there be more than a single milestone?

If not, I guess there's no need for the loop, and if it's possible, then the current code will accept the triage milestone if it's set with another one

**Review by chouquette — COMMENTED** [review]
A few nits but overall this looks good and seems like a very good idea! Thanks :)

File: .github/workflows/label-analysis.yml (hunk 1 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:19]: Yes, you are right, changing, thanks

File: .github/workflows/label-analysis.yml (hunk 2 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:52]: Hagrid

**Review by KevinFairise2 — APPROVED** [review]
Nice 🚀

**Pull request merged**